### PR TITLE
Fix float numbers for search on import

### PR DESCRIPTION
**Pull request by efirs**
This change detects if document contains array of objects in the schema, and if it is the case replaces all the numbers to fractional value, in order to guarantee that search infer their schema as float.

It inserts this record with replaced numbers for search to persists proper schema and deletes the record immediately.

It proceeds import process as usual after that

### CODE REVIEW

File: cmd/import.go [hunk 1]
```diff
@@ -54,6 +55,10 @@ var (
 	ErrCollectionShouldExist = fmt.Errorf("collection should exist to import CSV with no field names")
 	ErrNoAppend              = fmt.Errorf(
 		"collection exists. use --append if you need to add documents to existing collection")
+
+	ErrNoRecordsExpected = fmt.Errorf("no records expected in the collection after fixing numbers")
```
Comment (ovaistariq) [inline cmd/import.go:59]: Is this an error we will show to the user?

File: cmd/import.go [hunk 2]
```diff
@@ -74,7 +79,66 @@ func evolveSchema(ctx context.Context, db string, coll string, docs []json.RawMe
 	return util.Error(err, "create or update collection")
 }
 
+func fixNumbers(ctx context.Context, coll string, docs []json.RawMessage) {
+	err := schema.Infer(&sch, coll, docs[0:1], PrimaryKey, AutoGenerate, 1)
+	util.Fatal(err, "infer schema")
+
+	if schema.HasArrayOfObjects {
```
Comment (ovaistariq) [inline cmd/import.go:86]: Weill all these error messages be visible to the user? All of this has to be hidden and transparent to the user. We should not expose any of this we are doing to the user.

File: cmd/import.go [hunk 3]
```diff
@@ -74,7 +79,66 @@ func evolveSchema(ctx context.Context, db string, coll string, docs []json.RawMe
...
+		fixNumbers(ctx, coll, docs)
+
+		schema.DetectArrayOfObjects = false
+		schema.DetectIntegers = true
+		schema.ReplaceNumber = false
+
+		cnt, err := client.GetDB().Count(ctx, coll, driver.Filter("{}"))
+		util.Fatal(err, "get count after")
+
+		if cnt != 0 {
```
Comment (ovaistariq) [inline cmd/import.go:130]: Same thing here, all this fixing has to be transparent.

File: cmd/import.go (hunk 2 above)
Comment (ovaistariq) [inline cmd/import.go:86]: Is this issue only with `ArrayOfObjects` or does this affect `Objects` as well?

Comment (efirs) [reply cmd/import.go:86]: Looks like with array of objects only. @himank can confirm?

File: cmd/import.go (hunk 1 above)
Comment (efirs) [reply cmd/import.go:59]: This should not happen.

File: cmd/import.go (hunk 2 above)
Comment (efirs) [reply cmd/import.go:86]: Those are debug messages only shown in the debug log.

File: cmd/import.go (hunk 3 above)
Comment (efirs) [reply cmd/import.go:130]: Nothing is shown, only if there is error or debug log enabled.

File: cmd/import.go (hunk 2 above)
Comment (himank) [reply cmd/import.go:86]: That's right.

**Review by himank — APPROVED** [review]

**Pull request merged**